hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java

```diff
@@ -76,6 +78,11 @@
   private final boolean isInitialized;
   private ExecutorService installSnapshotExecutor;
 
+  // The atomic variable RaftServerImpl#inProgressInstallSnapshotRequest
+  // ensures serializable between notifyInstallSnapshotFromLeader()
+  // and reinitialize().
+  private DBCheckpoint installingDBCheckpoint;
+
```

Comment:
Once, the installSnaoshot completes, i guess, we need to set the "IntsallingCheckpoint" filed to null. Also, before the downloadSnaoshot icall is invoked, we should add a precondition check that installingCheckpoint is null always.